terraform/authentik-config/provider-app.tf

```diff
@@ -0,0 +1,54 @@
+
+resource "authentik_application" "coder" {
+  name               = "coder"
+  slug               = "coder"
+  group              = "ii"
+  meta_description   = "Coder for Sharing"
+  meta_icon          = "https://avatars.githubusercontent.com/u/95932066?s=200&v=4"
+  meta_launch_url    = "https://coder.${var.domain}/api/v2/users/oidc/callback?redirect=%2F"
+  open_in_new_tab    = false
+  policy_engine_mode = "any"
+  protocol_provider  = authentik_provider_oauth2.coder.id
+}
+
+
+data "authentik_scope_mapping" "coder" {
+  # Search by name, by managed field or by scope_name
+  # name    = "authentik default OAuth Mapping: Proxy outpost"
+  managed_list = [
+    "goauthentik.io/providers/oauth2/scope-email",
+    "goauthentik.io/providers/oauth2/scope-offline_access",
+    "goauthentik.io/providers/oauth2/scope-openid",
+    "goauthentik.io/providers/oauth2/scope-profile"
+  ]
+}
+
+data "authentik_flow" "default-provider-authorization-implicit-consent" {
+  slug = "default-provider-authorization-implicit-consent"
+}
+
+data "authentik_flow" "default-authentication-flow" {
+  slug = "default-authentication-flow"
+}
+
+resource "authentik_provider_oauth2" "coder" {
+  name               = "coder"
+  client_type        = "confidential" # OR public
+  client_id          = var.authentik_coder_oidc_client_id
+  client_secret      = var.authentik_coder_oidc_client_secret
```

Comment:
I'm having trouble generating these and correctly applying them to both coder and authentik to connect them. Opening a support request with Authentik, since we are using the goauthentik terraform provider.